### PR TITLE
implemented and added Loading animation for search results page

### PR DESCRIPTION
**Pull request by lifeeeDEV**
# Infinity Scroll using animated dots at the end of the search page during loading the new page load.

Changes:

Added src/components/LoadingIndicator.tsx
   - Loading animation

Made changes to src/screens/search_results/SearchResultsScreen.tsx
   - Implemented Loading animation

## Checklist

- [x] Screenshots added for any UX changes
- [x] Demos attached for animations/interactions
- [ ] Pointed at proper branch (`develop` not `main`)
- [x] Assigned PR to yourself
- [ ] Requested review from code owners
- [ ] Relevant labels added (`feature`, `documentation`, etc.)
- [x] `Closes #<issue-number>` added if this resolves a GitHub issue
- [x] Branch is up to date with develop, ran `git rebase develop` if necessary
- [x] All GitHub Actions are passing
- [ ] `npm run todo-ci` ran if any todo comments were created

### CODE REVIEW

**Comment by Thenlie** [comment]
Demo: https://www.youtube.com/watch?v=oKBWUnyRhNE

**Review by Thenlie — COMMENTED** [review]
Looking good so far! A couple of comments and questions.

**Comment by Thenlie** [comment]
https://github.com/Thenlie/Streamability/actions/runs/9248464020/job/25467390629?pr=948
There are a number of linter failures here. It looks like these are probably due to a system config mismatch between us. I would suggest running `npm run lint` rather than letting your editor format. Let me know if this doesn't resolve the issue.

**Comment by Thenlie** [comment]
Just noticed, this is pointed at `main` when it should be pointed at `develop`. You can update the base branch from this PR.

File: src/screens/search_results/SearchResultsScreen.tsx [hunk 1]
```diff
@@ -86,17 +96,15 @@ const SearchResultsScreen: React.FC = () => {
...
             />
-            {cards}
-            <Button
-                title='Load More'
-                loading={dataLoading}
-                onClick={refetch}
-                disabled={!moreToFetch}
-                sx={{ display: moreToFetch ? 'block' : 'none', marginBottom: 2 }}
-            />
+            <div className='relative w-full'>
```
[the remark this answers is not in the document]
Comment (lifeeeDEV) [reply src/screens/search_results/SearchResultsScreen.tsx:99]: No, you`re right! It isn't. I was just experimenting

File: src/components/LoadingIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            margin: '20px 0',
+        },
+        dot: {
+            width: '16px',
+            height: '16px',
+            margin: '0 4px',
+            backgroundColor: '#333',
+            borderRadius: '50%',
+            animation: 'dot-blink 1.4s infinite both',
+        },
```
[the remark this answers is not in the document]
Comment (lifeeeDEV) [reply src/components/LoadingIndicator.tsx:18]: I agree, it was just my first intention to make it like that for devs to understand. sorry

File: src/screens/search_results/SearchResultsScreen.tsx [hunk 3]
```diff
@@ -1,23 +1,16 @@
...
 
 const LOG = new Logger('SearchResultsScreen');
 
-/**
- * This loader is mostly built straight from the react-router docs
- * https://reactrouter.com/en/main/components/form#get-submissions
- *
- * @param request | HTTP GET request from the SearchInput component
- * @returns {Promise<string>} | the users query
- */
```
[the remark this answers is not in the document]
Comment (lifeeeDEV) [reply src/screens/search_results/SearchResultsScreen.tsx:18]: It was like, i have a lot of experience, my bad. I didn't take into account others who might not understand the whole thing!

File: src/components/LoadingIndicator.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Thenlie) [reply src/components/LoadingIndicator.tsx:18]: No problem! Let's update this here, but I do agree with you that your method is easier to understand. We could consider an effort to remove tailwind and move to this approach, but I think consistency across the project is important.

File: src/components/LoadingIndicator.tsx
[the remark this answers is not in the document]
Comment (lifeeeDEV) [reply src/components/LoadingIndicator.tsx:1]: Seems also very efficient! I really don't mind if we use mine or tailwinds loading circle:) My point was just to provide a Infinity Scroll without the button. Do it's completely fine to use a framework to do that ;)

**Pull request closed without merging**